### ISSUE TITLE
Deciding how much to give away in a bounty

### ISSUE THREAD

**Issue by vdeturckheim** [nodejs/security-wg#304]
@MylesBorins raised an issue regarding the Node.js core HackerOne program: We do not have a definitive way to decide the ammount of money we should give as bounty for a specific issue.

I'll write down my ideas later today.

@reedloden is there any standard in the IBB right now?

**Comment by MylesBorins** [comment]
I ended up rewarding the first bounty based on suggested defaults ($500) and realized that we don't have any guidelines for this. Would be good to have some sort of consistency

**Comment by reedloden** [comment]
The specific pre-approved amounts are listed in the table at the bottom of https://hackerone.com/nodejs. If the current impacts don't map well to how Node.js categorizes severity, can definitely change that to match so that it's not a surprise to the researcher as to how much a report will earn.

Impact | Description | Amount
-- | -- | --
High | Demonstrate that remote exploitation of this bug can be easily, actively, and reliably achieved. | $1,500+
Medium | Demonstrate that remote exploitation of this bug is very likely (e.g. good control a register). | $1,000
Minimum | Demonstrate the presence of a security bug with probable remote exploitation potential. | $500

Keep in mind the following as well:
> Only critical vulnerabilities that demonstrate complete compromise of the system's integrity or confidentiality are eligible for a bounty - typically Arbitrary Code Execution or equivalent impact.

If you want to award over $1,500 for an issue, please get IBB Panel approval first.

Does this help?

**Comment by MylesBorins** [comment]
> Only critical vulnerabilities that demonstrate complete compromise of the system's integrity or confidentiality are eligible for a bounty - typically Arbitrary Code Execution or equivalent impact.

Oops... that definitely was not the case for the vuln I just gave a bounty to. Is this something we need to fix?

**Comment by reedloden** [comment]
@MylesBorins Talking about https://hackerone.com/reports/340580, I assume?

Note the following as well:

> Only vulnerabilities in Node.js core will be considered for eligibility. Submissions related to nodejs.org and other project websites are explicitly NOT eligible.

Just that alone means that report isn't eligible...

Let me see if we can still cancel it. Depends on if it has been automatically paid out already.

**Comment by mhdawson** [comment]
I thought they were reviewed IBB Panel  before being awarded, but I guess that is only for larger amounts (and only through manual review from what I see above).  I think we should have a process that requires N people to approve an award (N to be decided).

**Comment by vdeturckheim** [comment]
Agreed with @mhdawson here. Also, we should get into a framework to formalize how much a given vuln is worth based on metrics. I am not certain how this can be done. Probably using CVSS and adding a majoration given which subsystem are impacted.

**Comment by vdeturckheim** [comment]
@indutny 
>Only critical vulnerabilities that demonstrate complete compromise of the system's integrity or confidentiality are eligible for a bounty - typically Arbitrary Code Execution or equivalent impact.

I think DoS fit here.

**Comment by indutny** [comment]
cc @reedloden maybe/

**Comment by reedloden** [comment]
Basically, if we're talking about the CIA triad (Confidentiality, Integrity, and Availability), then Availability issues are generally out-of-scope for IBB bounties, especially for projects that haven't had bounties before.

Not saying DoS vulnerabilities are not important, but there's a big difference between the ability to run arbitrary code and causing a process to crash or hang. The IBB wants to use its (limited) funding as best as possible to incentivize focus on security issues that would have the largest impact.

**Comment by indutny** [comment]
That's fine, I just wanted to clarify. Thank you, @reedloden !

**Comment by lirantal** [comment]
We raised this topic in the last agenda meeting and @vdeturckheim will take a look at that

**Comment by 0xdade** [comment]
Glad to see this discussion happening, as the person who got accidentally paid out to :)

**Comment by olihough86** [comment]
I'm seeing a lot of talk of Denial of Service here, but didn't he manage to hijack registry.nodejs.com, that's not denial of service...

I agree that this is out of scope but don't sit there downplaying this, this could have bee a disaster if a bad actor started serving out altered packages.

The researcher states he was getting legitimate NPM traffic.

**Comment by lirantal** [comment]
@olihough86 I think you mixed things.
The ineligibility for the bounty isn't related to DoS, but to the fact that the website itself is out of scope.

From the policy mentioned on the H1 page:
```
Only vulnerabilities in Node.js core will be considered for eligibility. Submissions related to nodejs.org and other project websites are explicitly NOT eligible.
```

**Comment by olihough86** [comment]
I guess I got the two conversations in this thread mixed up :D
As I said I agree it's out of scope

**Comment by 0xdade** [comment]
For what it's worth, I was receiving what I am sure is a comparatively minimal amount of traffic. I gained access of the domain on April 3rd and reached out to the security@nodejs dot org email address, after a while of no response I was made aware that the nodejs hackerone program was made active, so I submitted it there. I controlled the domain from Apr 3 -> Apr 29 and only received requests for roughly 600 packages. I 404'd all of them with a 0 byte response. I am not sure if registry.nodejs.org was ever meant to be an actual registry, but there were a couple guides on the internet telling people to use it.

Anyways - it's cool that the takeover wasn't in scope, I'm a little bummed to see the bounty get revoked but I understand completely. Glad I was able to help out before someone else found it, at least :)
